go/registry/api/api.go

```diff
@@ -930,6 +930,34 @@ func VerifyRegisterRuntimeArgs(logger *logging.Logger, sigRt *SignedRuntime, isG
 	return &rt, nil
 }
 
+// VerifyRegisterComputeRuntimeArgs verifies ComputeRuntime-specific arguments for RegisterRuntime.
+func VerifyRegisterComputeRuntimeArgs(logger *logging.Logger, rt *Runtime, km *Runtime) error {
+	// Check runtime's key manager, if key manager ID is set.
+	if rt.KeyManager != nil {
+		// key manager runtime must exist.
+		if km == nil {
```

Comment:
This seems awkward, as mentioned in my previous review, it would be better to handle this via a `RuntimeLookup` interface, similar to the [existing `NodeLookup` interface](https://github.com/oasislabs/oasis-core/blob/master/go/registry/api/api.go#L274-L282). I will also need this interface in #2504.

So instead of passing `km`, pass the `RuntimeLookup` interface and call `runtimeLookup.Runtime(*rt.KeyManager)` here.

Reply:
I added `RuntimeLookup` with only single method `Runtime()` now. This method was already implemented before by `ImmutableState`.